api/api_sources/sources/server/modules/request.access/requestAccess.routes.ts

```diff
@@ -153,32 +160,46 @@ class RequestAccessRouteController extends RouteController {
             accessRequest.requestedAccessCode = req.body.requestedAccessCode;
         }
         // Check status
-        let requireDel = false;
         if (req.body.status && req.body.status !== accessRequest.status) {
-            [accessRequest, requireDel] = await this.handleStatusUpdate(req.body.status, accessRequest, req.user);
-        }
-        // Save or delete
-        if (requireDel) {
-            await this.dataController.remove(accessRequest);
-        } else {
-            await this.dataController.saveInDB(accessRequest);
+            accessRequest = await this.handleStatusUpdate(req.body.status, accessRequest, req.user);
         }
 
+        await this.dataController.saveInDB(accessRequest);
+
         return [200, accessRequest];
     }
 
     @Post({
         path: '/',
         middleware: () => [
             ValidatorExists({
-                requestedAccessCode: RoleCodeController.shared
+                requestedAccessCode: RoleCodeController.shared,
             }),
             check('requestNote').isString(),
             sanitize('requestNote')
         ]
     })
     public async create(req: any) {
         const input = req.body as CreateRequestAccess;
+        const requestedRole = input.requestedAccessCode;
+        const currentRoles: RolesCode[] = req.user.roles;
+
+        const isExistingRole = currentRoles.find(role => role.role_code_id === requestedRole.role_code_id);
+
+        if (isExistingRole) {
+            return [400, `You already have ${requestedRole.role} access`, true];
+        }
+
+        const hasPendingAccessRequest = await this.dataController.fetchOne({
+            requester: req.user,
+            requestedAccessCode: requestedRole,
+            status: 0
+        });
+
+        if (hasPendingAccessRequest) {
```

Comment:
400 is not appropriate here. Use 200

Reply:
Shouldn't this be a 400 since it's an error?

Reply:
Oh, you're saying that we just make it as an info for the users. In that case, I think it makes sense to return 200